src/core/agents.jl

```diff
@@ -1,54 +1,68 @@
-export AbstractAgent, @agent, GraphAgent, GridAgent, ContinuousAgent, OSMAgent
+export AbstractAgent, @agent, NoSpaceAgent
 
 """
     YourAgentType <: AbstractAgent
 Agents participating in Agents.jl simulations are instances of user-defined Types that
-are subtypes of `AbstractAgent`. It is almost always the case that mutable Types make
-for a simpler modelling experience.
+are subtypes of `AbstractAgent`.
 
-Your agent type(s) **must have** the `id` field as first field.
-Depending on the space structure there might be a `pos` field of appropriate type
-and a `vel` field of appropriate type.
-Each space structure quantifies precisely what extra fields (if any) are necessary,
-however we recommend to use the [`@agent`](@ref) macro to help you create the agent type.
+Your agent type(s) **must have** the `id::Int` field as first field.
+In Julia versions ≥ v1.8, this must also be declared as a `const` field.
+If any space is used (see [Available spaces](@ref)), a `pos` field of appropriate type
+is also mandatory. Each space may also require additional fields that may,
+or may not, be communicated as part of the public API.
 
-Your agent type may have other additional fields relevant to your system,
-for example variable quantities like "status" or other "counters".
-
-As an example, a [`GraphSpace`](@ref) requires an `id::Int` field and a `pos::Int` field.
-To make an agent with two additional properties, `weight, happy`, we'd write
-```julia
-mutable struct ExampleAgent <: AbstractAgent
-    id::Int
-    pos::Int
-    weight::Float64
-    happy::Bool
-end
-```
+The [`@agent`](@ref) macro ensures that all of these constrains are in place
+and hence it is the expected way to generate new agent types.
 """
 abstract type AbstractAgent end
 
 """
-    @agent YourAgentType{X, Y} AgentSupertype begin
-        some_property::X
-        other_extra_property::Y
+    @agent YourAgentType{X} AnotherAgentType [OptionalSupertype] begin
+        extra_property::X
+        other_extra_property::Int
         # etc...
     end
 
-Create a struct for your agents which includes the mandatory fields required to operate
-in a particular space. Depending on the space of your model, the `AgentSupertype` is
-chosen appropriately from [`GraphAgent`](@ref), [`GridAgent`](@ref),
-[`ContinuousAgent`](@ref), [`OSMAgent`](@ref).
+Define an agent struct which includes all fields that `AnotherAgentType` has,
+as well as any additional ones the user may provide via the `begin` block.
+See below for examples.
+
+Using `@agent` is **the recommended way to create agent types** for using in Agents.jl.
```

Comment:
Similar to above, I feel like mentioning that it is the recommended way leaves the question that there are other ways that we support, which isn't the case.